src/ORM/DataQuery_SubGroup.php

```diff
@@ -14,21 +14,39 @@
  */
 class DataQuery_SubGroup extends DataQuery implements SQLConditionGroup
 {
+    private string $clause;
 
     /**
-     *
      * @var SQLSelect
      */
     protected $whereQuery;
 
+    /**
+     * @var SQLSelect
+     */
+    protected $havingQuery;
+
     public function __construct(DataQuery $base, $connective)
     {
+        // using func_get_args to add a 3rd param while retaining BC
+        // @deprecated - add a 3rd param for CMS 6 - string $clause = 'WHERE'
+        $clause = 'WHERE';
+        $args = func_get_args();
+        if (count($args) > 2) {
+            $clause = $args[2];
+        }
         parent::__construct($base->dataClass);
         $this->query = $base->query;
-        $this->whereQuery = new SQLSelect();
-        $this->whereQuery->setConnective($connective);
-
-        $base->where($this);
+        $this->clause = strtoupper($clause);
+        if ($this->clause === 'WHERE') {
+            $this->whereQuery = new SQLSelect();
+            $this->whereQuery->setConnective($connective);
+            $base->where($this);
+        } elseif ($this->clause === 'HAVING') {
+            $this->havingQuery = new SQLSelect();
+            $this->havingQuery->setConnective($connective);
+            $base->having($this);
+        }
```

Comment:
Need a final `else` - probably throw an exception since we're not expecting it.
Otherwise, if there's a compelling reason to not throw an exception, there should at least be a comment documenting the reason.

Reply:
Updated